lib/rendition.js

```diff
@@ -42,17 +102,158 @@ class Rendition {
     }
 
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
+    }
+
+    async sha1() {
+        if (!this._sha1) {
+            this._sha1 = await fileHash(this.path, HASH_ALGORITHM);
+        }
+        return this._sha1;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    exists() { // see and read the file
+        return fs.existsSync(this.path);
+    }
+
+    async setMimeInformation(mime=null, encoding=null){
+        // for javascript workers mostly ------------------------------------------------------
+        // explicitly set mimetype would get priority
+        if(mime){ // to handle case when this._mime is set to default DEFAULT_CONTENT_TYPE
+            this._mime = mime;
+        }
+
+        // explicitly set encoding would get priority
+        if(encoding){ // to handle case when this.encoding would have a default value
+            this._encoding = encoding;
+        }
+
+        // shellscript mostly ------------------------------------------------------
+        if(this.mimeInfoPath && await fs.pathExists(this.mimeInfoPath)){ // use the path containing the file with mime information
+            let mimeInfoContent = await fs.readFile(this.mimeInfoPath);
+            mimeInfoContent = mimeInfoContent.toString();
+            const mimeInfo = mimeInfoContent.split(";");
+
+            if(mimeInfo && mimeInfo.length === 2){ // we got something in an expected format
+                this._mime = mime || mimeInfo[0].trim() || DEFAULT_CONTENT_TYPE;
+                this._encoding = encoding || mimeInfo[1].trim();
+            }
+        } 
+        // fallback if things are missing ------------------------------------------------------
+        if(!this._encoding || !this._mime){ // fallback to using file-type lib (won't set encoding)
+            try { 
+                await commandExists('file'); // throws if `file` is not found
```

Comment:
Some context: My idea by checking was to avoid running a "useless" (failing) exec. So I double check that lib's code and it falls back anyway to trying it with exec.

Also, we're sure we only need this working on *nix, no WIndows support needed for this for now. 

Because of those 2 reasons, agreed, I'll remove the commandExists check.